core/src/main/java/org/elasticsearch/index/indexing/ShardIndexingService.java

```diff
@@ -124,19 +128,31 @@ public void postCreate(Engine.Create create) {
             try {
                 listener.postCreate(create);
             } catch (Exception e) {
-                logger.warn("post listener [{}] failed", e, listener);
+                logger.warn("postCreate listener [{}] failed", e, listener);
             }
         }
     }
 
     public void postCreate(Engine.Create create, Throwable ex) {
+        for (IndexingOperationListener listener : listeners) {
+            try {
+                listener.postCreate(create, ex);
+            } catch (Exception e) {
+                logger.warn("postCreate listener [{}] failed", e, listener);
+            }
+        }
     }
 
     public Engine.Index preIndex(Engine.Index index) {
         totalStats.indexCurrent.inc();
         typeStats(index.type()).indexCurrent.inc();
         for (IndexingOperationListener listener : listeners) {
-            index = listener.preIndex(index);
+            try {
+                listener.preIndex(index);
+            } catch (Exception e) {
```

Comment:
we shouldn't catch the exception here to be honest. I think in all the preIndex|Create|Delete we should just fail hard and bubble up the exception. It would be awesome if we could document that on the interface that failures in this method will be fatal and prevent the document from being indexed|created|deleted?